### PR TITLE
New R language Cheat sheet.

### PR DESCRIPTION
**Pull request by rojosewe**
https://duck.co/ia/view/r_language_cheatsheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@rojosewe Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=R Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/r_language_cheatsheet_mobile.png)

File: share/goodie/cheat_sheets/json/rcheat_sheet.json [hunk 1]
```diff
@@ -0,0 +1,96 @@
+{
+    "id": "rcheat_sheet_cheat_sheet",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/rcheat_sheet.json]: this must be `r_language_cheatsheet`

File: share/goodie/cheat_sheets/json/rcheat_sheet.json [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    "name": "RCheatSheet",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/rcheat_sheet.json]: lets remove cheatsheet from name

File: share/goodie/cheat_sheets/json/rcheat_sheet.json [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    "description": "R Programming language Syntax",
+
+    "metadata": {
+        "sourceName": "cran: R project",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/rcheat_sheet.json]: lets use `"sourceName": "cran: R project",`

File: share/goodie/cheat_sheets/json/rcheat_sheet.json [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                "key": "<- or =",
+                "val": "Assignment"
+            },
+            {
+                "key": "<<-",
+                "val": "Global Assignment"
+            },
+            {
+                "key": "v[1]",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/rcheat_sheet.json]: use `"key": "v\\[1\\]",`

File: share/goodie/cheat_sheets/json/rcheat_sheet.json [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                "key": "*",
+                "val": "Scalar Multiplication"
+            },
+            {
+                "key": "%*%",
+                "val": "Matrix Multiplication"
+            },
+            {
+                "key": "/",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/rcheat_sheet.json]: use  `"key": "\/",`

File: share/goodie/cheat_sheets/json/rcheat_sheet.json [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            }
+        ],
+        "Data creation": [
+            {
+                "key": "c(...)",
+                "val": "Forming a vector"
+            },
+            {
+                "key": "from:to",
+                "val": "Generates a sequence from a given number to another."
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/rcheat_sheet.json]: lets remove the dot  at the end

**Comment by gautamkrishnar** [comment]
@rojosewe Welcome to DuckDuckHack thanks for taking your time to contribute :smile:

File: share/goodie/cheat_sheets/json/rcheat_sheet.json [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    "name": "RCheatSheet",
+    "description": "R Programming language Syntax",
+
+    "metadata": {
+        "sourceName": "cran: R project",
+        "sourceUrl" : "https://cran.r-project.org/doc/contrib/Short-refcard.pdf"
+    },
+
+    "aliases": [
+        "R", "Rlang", "R language", "Cran", "CRAN R"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/rcheat_sheet.json]: @rojosewe avoid uppercase letters in aliases

File: share/goodie/cheat_sheets/json/rcheat_sheet.json [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                "key": "<<-",
+                "val": "Global Assignment"
+            },
+            {
+                "key": "v[1]",
+                "val": "First element in a vector"
+            },
+            {
+                "key": "v[1,]",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/rcheat_sheet.json]: @rojosewe is there a , after 1 or  is it a typo?

File: share/goodie/cheat_sheets/json/rcheat_sheet.json (hunk 7 above)
Comment (moollaza) [inline share/goodie/cheat_sheets/json/rcheat_sheet.json]: @rojosewe @gautamkrishnar  @pa1pal 

Both of your "R" Cheat Sheets have overlapping aliases which isn't allowed. Aliases must be unique to a single cheat sheet as only 1 can trigger. Also they must be lowercased. I think this cheat sheet deserves the "r", "rlang", "rlanguage", ... etc triggers. The "R Fucntions" Cheat Sheet is more specific.

Please agree on which cheat sheets will use each of these aliases.

File: share/goodie/cheat_sheets/json/rcheat_sheet.json (hunk 1 above)
Comment (moollaza) [inline share/goodie/cheat_sheets/json/rcheat_sheet.json]: also @rojosewe the filename should match the id, with `cheat_sheet` removed, and `_` replaced with `-`

i.e. `r-language.json`

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=R Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/r_language_cheatsheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=R Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/r_language_cheatsheet_mobile.png)

**Comment by moollaza** [comment]
@rojosewe Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=R Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/r_language_cheatsheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=R Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/r_language_cheatsheet_mobile.png)

**Comment by moollaza** [comment]
@rojosewe I'm going to close this for now. If you'd like to finish it up, please let us know and we'd be happy to reopen this for you!

**Comment by gautamkrishnar** [comment]
@moollaza i will be happy to continue this. Lets wait a few more days....

**Comment by moollaza** [comment]
@rojosewe thanks again for all the hard work you've put into creating this R Language Cheat Sheet. You've done a great job so far! We at DuckDuckGo, and the DuckDuckHack community really appreciate you taking the time to contribute.

We'd love to see it go live and it looks like it's mostly done?  If you're stuck or need any help, I'd love to chat if you're around this week or next. Please message me on slack (@moollaza) or send me an email (moollaza@duckduckgo.com) if you'd be interested in talking. 

Thanks again!

**Comment by pjhampton** [comment]
@gautamkrishnar @moollaza Hey guys

Has this been finished off? If not, I can sort it out tomorrow - I use R daily as part of my research.

Thanks

**Pull request closed without merging**